modules/RTC/TPCUtils.js

```diff
@@ -34,19 +35,93 @@ export class TPCUtils {
+            for (const codec of VIDEO_CODECS) {
+                const bitrateSettings = videoQualitySettings[codec]?.maxBitratesVideo
+                    ?? videoQualitySettings[codec.toUpperCase()]?.maxBitratesVideo
+                    ?? (videoQualitySettings.maxBitratesVideo
+                        && (videoQualitySettings.maxBitratesVideo[codec]
+                            || videoQualitySettings.maxBitratesVideo[codec.toUpperCase()]));
+                const scalalabilityModeSettings = videoQualitySettings[codec]
+                    ?? videoQualitySettings[codec.toUpperCase()];
+
+                if (bitrateSettings) {
+                    [ 'low', 'standard', 'high', 'ssHigh' ].forEach(value => {
+                        if (bitrateSettings[value]) {
+                            this.codecSettings[codec].maxBitratesVideo[value] = bitrateSettings[value];
+                        }
+                    });
+                }
+
+                const scalabilityModeEnabled = this.codecSettings[codec].scalabilityModeEnabled
+                    && (typeof scalalabilityModeSettings?.scalabilityModeEnabled === 'undefined'
+                        || scalalabilityModeSettings.scalabilityModeEnabled);
+
+                if (scalabilityModeEnabled) {
+                    scalalabilityModeSettings?.useSimulcast
+                        && (this.codecSettings[codec].useSimulcast = scalalabilityModeSettings.useSimulcast);
+                    scalalabilityModeSettings?.useL3T3
+                        && (this.codecSettings[codec].useL3T3 = scalalabilityModeSettings.useL3T3);
+                } else {
+                    this.codecSettings[codec].scalabilityModeEnabled = false;
+                }
+            }
+        }
+    }
+
+    /**
+     * Calculates the configuration of the first encoding when the client is configured to do full SVC for VP9 and AV1
+     * codecs depending on the resolution that is being requested.
+     *
+     * @param {JitsiLocalTrack} localVideoTrack - The local video track.
+     * @param {number} newHeight - The resolution that needs to be configured for the local video track.
+     * @param {CodecMimeType} codec - The video codec.
+     * @returns {Object} configuration.
+     */
+    _calculateActiveEncodingParamsForSvc(localVideoTrack, newHeight, codec) {
+        const codecBitrates = this.codecSettings[codec].maxBitratesVideo;
+        const localTrack = localVideoTrack.getTrack();
+        const { height } = localTrack.getSettings();
+        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopBitrate || DESKTOP_SHARE_RATE;
+        const isScreenshare = localVideoTrack.getVideoType() === VideoType.DESKTOP;
+        const scalabilityMode = isScreenshare && this._isScreenshareBitrateCapped(localVideoTrack)
+
+            // The endpoint needs to send only 1 spatial layer with temporal scalability for low fps SS.
+            ? VideoEncoderScalabilityMode.L1T3
```

Comment:
Please rewirte this to not use nested inline conditionals, it's very hard to read and easy to make mistakes.